src/averageAges.js

```diff
@@ -37,7 +45,17 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const milfs = people.filter(milf => {
+    return (withChildren === true)
+      ? (people.some(hasThisMother => hasThisMother.mother === milf.name))
+      : milf.sex === 'f';
+  });
+
+  const sumFilteredAges = milfs.reduce((sum, { born, died }) => {
+    return sum + (died - born);
+  }, 0);
+
+  return sumFilteredAges / milfs.length;
```

Comment:
you repeat the same logic three times, so let us create a callback for these calculations